adafruit_bno055.py

```diff
@@ -19,11 +19,150 @@
+    +------------------+---------------+--------------+-----------+----------+
+    | GYRONLY_MODE     |       -       |       -      |     X     |     -    |
+    +------------------+---------------+--------------+-----------+----------+
+    | ACCMAG_MODE      |       X       |       X      |     -     |     -    |
+    +------------------+---------------+--------------+-----------+----------+
+    | ACCGYRO_MODE     |       X       |       -      |     X     |     -    |
+    +------------------+---------------+--------------+-----------+----------+
+    | MAGGYRO_MODE     |       -       |       X      |     X     |     -    |
+    +------------------+---------------+--------------+-----------+----------+
+    | AMG_MODE         |       X       |       X      |     X     |     -    |
+    +------------------+---------------+--------------+-----------+----------+
+    | IMUPLUS_MODE     |       X       |       -      |     X     |     -    |
+    +------------------+---------------+--------------+-----------+----------+
+    | COMPASS_MODE     |       X       |       X      |     -     |     X    |
+    +------------------+---------------+--------------+-----------+----------+
+    | M4G_MODE         |       X       |       X      |     -     |     -    |
+    +------------------+---------------+--------------+-----------+----------+
+    | NDOF_FMC_OFF_MODE|       X       |       X      |     X     |     X    |
+    +------------------+---------------+--------------+-----------+----------+
+    | NDOF_MODE        |       X       |       X      |     X     |     X    |
+    +------------------+---------------+--------------+-----------+----------+
+
+    The default mode is ``NDOF_MODE``.
+    """
+
+    temperature = _ReadOnlyUnaryStruct(0x34, 'B')
+    """Measures the temperature of the chip in degrees Celsius."""
+    accelerometer = _ScaledReadOnlyStruct(0x08, '<hhh', 1/100)
+    """Gives the raw accelerometer readings, in m/s."""
+    magnetometer = _ScaledReadOnlyStruct(0x0e, '<hhh', 1/16)
+    """Gives the raw magnetometer readings in microteslas."""
+    gyroscope = _ScaledReadOnlyStruct(0x14, '<hhh', 1/900)
+    """Gives the raw gyroscope reading in degrees per second."""
+    euler = _ScaledReadOnlyStruct(0x1a, '<hhh', 1/16)
+    """Gives the calculated orientation angles, in degrees."""
+    quaternion = _ScaledReadOnlyStruct(0x20, '<hhhh', 1/(1<<14))
+    """Gives the calculated orientation as a quaternion."""
+    linear_acceleration = _ScaledReadOnlyStruct(0x28, '<hhh', 1/100)
+    """Returns the linear acceleration, without gravity, in m/s."""
+    gravity = _ScaledReadOnlyStruct(0x2e, '<hhh', 1/100)
+    """Returns the gravity vector, without acceleration in m/s."""
+
+    def __init__(self, i2c, address=0x28):
+        self.i2c_device = I2CDevice(i2c, address)
+        self.init()
+
+    def init(self, mode=NDOF_MODE):
+        chip_id = self._chip_id  # WARNING! Side effects!
```

Comment:
These property reads don't actually have side effects. They just read state that may change. You are also welcome to change the internal properties into manual i2c read and writes if you think that will make it clearer. The register library is best used for directly exposed registers.